src/adapters/supabase/helpers/user.ts

```diff
@@ -0,0 +1,63 @@
+import { Database } from "../types/database";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Super } from "./supabase";
+import { Context } from "../../../types/context";
+import { addCommentToIssue } from "../../../utils/issue";
+
+export class User extends Super {
+  constructor(supabase: SupabaseClient<Database>, context: Context) {
+    super(supabase, context);
+  }
+
+  async getWalletByUserId(userId: number, issueNumber: number) {
+    const { data, error } = await this.supabase.from("users").select("wallets(*)").eq("id", userId).single();
+    if ((error && !data) || !data.wallets?.address) {
+      this.context.logger.error("No wallet address found", { userId, issueNumber }, true);
+      if (this.context.config.miscellaneous.startRequiresWallet) {
+        await addCommentToIssue(this.context, "```diff\n! Please set your wallet address with the /wallet command first and try again.\n```");
+        throw new Error("No wallet address found");
+      } else {
+        await addCommentToIssue(this.context, "```diff\n# Please set your wallet address with the /wallet command in order to receive a task reward.\n```");
+      }
+    } else {
+      this.context.logger.info("Successfully fetched wallet", { userId, address: data.wallets?.address });
+    }
+
+    return data?.wallets?.address || null;
+  }
+
+  public async getMultiplier(userId: number, repositoryId: number) {
```

Comment:
@Keyrxng As far as I understand `getMultiplier()`, `_getAccessData()` and `getLocationsFromRepo()` are not used anywhere. If so pls remove them.

Reply:
my mistake, resolved.